lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the vault-autounseal relation.
+
+This library contains the Requires and Provides classes for handling the
+vault-autounseal interface.
+
+The provider side of the interface is responsible for enabling the vault
+transit engine and creating the necessary keys and policies for an external
+vault to be able to autounseal itself.
+
+The requirer side of the interface is responsible for retrieving the necessary
+details to autounseal the vault instance, and configuring the vault instance to
+use them.
+
+## Getting Started
+
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.vault_k8s.v0.vault_autounseal
+```
+
+### Provider charm
+
+The provider charm is the charm that provides a Vault instance that can be
+used to autounseal other Vault instances via the Vault transit backend.
+
+Add the following to `metadata.yaml`:
+
+```yaml
+provides:
+  vault-autounseal-provides:
+    interface: vault-autounseal
+```
+
+### Requirer charm
+
+The requirer charm is the charm that wishes to autounseal a Vault instance via
+the Vault transit backend.
+
+Add the following to `metadata.yaml`:
+
+```yaml
+requires:
+  vault-autounseal-requires:
+    interface: vault-autounseal
+    limit: 1
+```
+
+### Integration
+
+You can integrate both charms by running:
+
+```bash
+juju integrate <vault a>:vault-autounseal-provides <vault b>:vault-autounseal-requires
+```
+
+where `vault a` is the Vault app which will provide the autounseal service, and
+`vault b` is the Vault app which will be configured for autounseal via `vault a`.
+"""
```

Comment:
Do you plan to add some examples of provifer and requirer charms?

Reply:
No, I don't think so. This more of an internal lib. But we can do it if there is a good reason.

Reply:
Okay done in #499

Reply:
I added it and then decided to remove it again.
This lib like the `vault-kv` lib is really opinionated, and in addition this lib is meant to be used only with a Vault charm (Vault is both the requirer and the provider)
I prefer to keep it as is for now.